tests/ArrayCaster/ArrayV2CasterTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Spatie\DataTransferObject\Tests\ArrayCaster;
+
+use Spatie\DataTransferObject\DataTransferObject;
+use Spatie\DataTransferObject\Tests\TestCase;
+
+class ArrayV2CasterTest extends TestCase
+{
+    /** @test */
+    public function version_2_will_casted()
+    {
+        $bar = new ParentDto(
```

Comment:
Hi @brendt,

this is our actual specific error when upgrading to v3.

Thank you

Reply:
on our project, its too difficult to refactor the DTO, since many using on this object